core-bundle/src/Image/Studio/ImageResult.php

```diff
@@ -63,24 +69,51 @@ class ImageResult
      *
      * @internal Use the Contao\Image\Studio\Studio factory to get an instance of this class
      */
-    public function __construct(ContainerInterface $locator, string $projectDir, $filePathOrImage, $sizeConfiguration = null)
+    public function __construct(ContainerInterface $locator, string $projectDir, $filePathOrImage, $sizeConfiguration = null, ResizeOptions $resizeOptions = null)
     {
         $this->locator = $locator;
         $this->projectDir = $projectDir;
         $this->filePathOrImageInterface = $filePathOrImage;
         $this->sizeConfiguration = $sizeConfiguration;
+        $this->resizeOptions = $resizeOptions;
     }
 
     /**
      * Creates a picture with the defined size configuration.
      */
     public function getPicture(): PictureInterface
     {
-        if (null === $this->picture) {
-            $this->picture = $this->pictureFactory()->create($this->filePathOrImageInterface, $this->sizeConfiguration);
+        if (null !== $this->picture) {
+            return $this->picture;
+        }
+
+        // Unlike the Contao\Image\PictureFactory the PictureFactoryInterface
+        // does not know about ResizeOptions. We therefore check if the third
+        // argument of the 'create' method allows setting them.
+        $canHandleResizeOptions = static function (PictureFactoryInterface $factory): bool {
+            $createParameters = (new \ReflectionClass($factory))
```

Comment:
Using reflection in production code does not seem correct to me.

Reply:
I generally am not a big fan of reflection either. But this scenario is imo the intended use case for it (even in prod). Or do you know a better/easier way to check this?

Reply:
@ausi /cc

Reply:
I think maybe we could just don’t do the check because additional method parameters are valid in PHP. It would only break then if someone implements the method with a different third parameter. But I’m not sure.

Reply:
Hmm, I don't really like this. Imo, chances that I extend the signatures are pretty high if I'm already creating my own factory instead of using the core's default.

Reply:
I guess the only other strategy would probably be to add another `PictureFactoryResizeOptionsInterface` (that possibly extends the `PictureFactoryInterface`) and redefines (widens) the `create` method. And then we would check on this interface instead (and also apply it to our `PictureFactory`).

Reply:
Should we add a shortcut by checking `if ($factory instanceof PictureFactory) { return true; }`?
This way reflection is only used for custom picture factories.

Reply:
Good idea!

Implemented it in 0d1e22cd76c486e3cf5bb7df5a089ccda1f8f851.

Reply:
You could also simplify this with a `createWithResize` method and a `method_exists` check.

Reply:
If everyone would need to add this method in their implementation shouldn't this be an interface then? 

`PictureFactoryWithResizeOptionsInterface extends PictureFactoryInterface` or sth

Reply:
I think we should keep it as it is in this pull request currently. 

But if we want to get rid of the reflection, we should create a new interface and deprecate the old one. Using a seperate method doesn’t make sense IMO.